maths/series/arithmetico_geometric_sequence.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+"""
+This is a pure Python implementation of the Arithmetic–Geometric Sequence algorithm
+https://en.wikipedia.org/wiki/Arithmetico%E2%80%93geometric_sequence
+For doctests run following command:
+python -m doctest -v arithmetic_geometric_sequence.py
+or
+python3 -m doctest -v arithmetic_geometric_sequence.py
+For manual testing run:
+python3 arithmetic_geometric_sequence.py
+"""
+
+
+class AGS:
+    def __init__(self, a, d, b, r, n):
```

Comment:
Needs type hints.

Single-letter variables are old school.  What about self-documenting parameter names like arithmetic_value, arithmetic_difference, geometric_value, geometric_difference, term_count, k_in_kth_term?  If you can come up with even better names that would be great.